addon/ng2/tasks/format.js

```diff
@@ -0,0 +1,23 @@
+/* jshint node: true */
+'use strict';
+
+var Promise = require('ember-cli/lib/ext/promise');
+var Task = require('ember-cli/lib/models/task');
+var exec = Promise.denodeify(require('child_process').exec);
```

Comment:
I suppose, in the case where they are first imported here, it would be faster, yes.

Leaving them at the top seems to be the normal usage pattern, and it's also a bit more forwards compatible with the ES6 mindset. I'm also not a big fan of conditional imports myself, they're sometimes hard to find, test and debug.

Come to think of it, I remember reading they opted with toplevel imports for a number of reasons that were meant to get around problems with node's modules system.

I don't see much advantage in this case to only requiring them when the task is run.